### PR TITLE
Adding an option for trailing stoploss: "trailing_only_offset_is_reached"

### PR DESCRIPTION
**Pull request by mishaker**
## Summary
This will permit to activate trailing stoploss only if the offset is reached. So basically only the part of the trailing stoploss after offset will be used. 

## TODO 
- [X] Fixing current tests
- [X] Adding new tests for the option
- [X] Add config validation
- [X] Documentation
- [X] Test on live

### CODE REVIEW

File: freqtrade/resolvers/strategy_resolver.py [hunk 1]
```diff
@@ -46,18 +46,19 @@ def __init__(self, config: Optional[Dict] = None) -> None:
...
-                      ("trailing_stop_positive",        None,    False),
-                      ("trailing_stop_positive_offset", 0.0,     False),
-                      ("process_only_new_candles",      None,    False),
-                      ("order_types",                   None,    False),
-                      ("order_time_in_force",           None,    False),
-                      ("use_sell_signal",               False,   True),
-                      ("sell_profit_only",              False,   True),
-                      ("ignore_roi_if_buy_signal",      False,   True),
+        attributes = [("minimal_roi", None, False),
+                      ("ticker_interval", None, False),
```
Comment (xmatthias) [inline freqtrade/resolvers/strategy_resolver.py]: mmmh ... the "old" format was easier to read ...

File: freqtrade/exchange/exchange.py [hunk 2]
```diff
@@ -257,6 +259,28 @@ def validate_order_time_in_force(self, order_time_in_force: Dict) -> None:
                 raise OperationalException(
                     f'Time in force policies are not supporetd for  {self.name} yet.')
 
+    def validate_trailing_stoploss(self, config) -> None:
```
Comment (xmatthias) [inline freqtrade/exchange/exchange.py]: I really don't think that this should be in the exchange class, but rather in configuration, or in a seperate "validation" file.

it's got nothing to do with the exchange (which is different in the other validate functions ...) ...

**Comment by mishaker** [comment]
@xmatthias do you see any reason this PR can be related to process exit without any message ?

**Comment by xmatthias** [comment]
no - but I've also never really seen that ... normally you always see "something" when it exits, unless the process is killed maybe ...

**Pull request merged**